### ISSUE TITLE
Base non generic HtmlHelpers class for Nancy.ViewEngines.Razor

### ISSUE THREAD

**Issue by mikle111** [NancyFx/Nancy#2073]
There is some thing in Nancy Razor realisation that makes me unhappy:
If we want IntelliSense to work with our view than we need to write 
```csharp
@inherits Nancy.ViewEngines.Razor.NancyRazorViewBase<ModelType>
```
Than I want to use some helpers. All works just ok for simple model class. But if I have generic like ISomeInterface<ISomethingElse> than helpers doesn't work.
So it can be fixed with base non generic class(or interface) for helpers that we can use in extension methods. Its simple, and seems logical. That class can contain model independent fields and methods like:
```csharp
public class HtmlHelpers 
{
    RazorViewEngine Engine { get; set; }
    IRenderContext RenderContext { get; set; }
    IHtmlString Partial(string viewName){}
    IHtmlString Raw(string text){}
    //....and etc.
}
```

**Comment by phillip-haydon** [comment]
Hmmm 

So you have models that have an interface? And you want helpers that work against the interface?

**Comment by mikle111** [comment]
Oh, I missed up some type parameters for my extension methods. It works now.
There is still a problem. Assume that I have model independent html helper extension:
```csharp
public static INamedData<T, TModel> GetNamedData<T>(this HtmlHelpers<TModel> html, string name = null)
        {
            return html.RenderContext.Context.GetClientEnvironment().GetNamedData<T>(name ?? typeof(T).Name);
        }
```
But I have to use `TModel` parameter. And if I'll use that extension frome some other place, I'll need to add `TModel` type parameter here to. Why use generics, when dont need them? There is model independent part in HtmlHelpers class, that can be (and should be by single responsibility principle) separeted without any significant efforts.
I have made pull request with code that solves it https://github.com/NancyFx/Nancy/pull/2074
So extension will be like:
```csharp
public static INamedData<T> GetNamedData<T>(this HtmlHelpers html, string name = null)
        {
            return html.RenderContext.Context.GetClientEnvironment().GetNamedData<T>(name ?? typeof(T).Name);
        }
```
Unusable type parameter that flows into every place with helpers has gone. Im happy)

**Comment by khellang** [comment]
Closed with #2074
